src/java/test/org/apache/zookeeper/server/ZooKeeperServerStartupTest.java

```diff
@@ -0,0 +1,266 @@
+                    simplezks.waitForSessionCreation(5));
+
+            LOG.info(
+                    "Decrements the count of the latch, so that server will proceed with startup");
+            startupDelayLatch.countDown();
+
+            Assert.assertTrue("waiting for server being up ", ClientBase
+                    .waitForServerUp(HOSTPORT, ClientBase.CONNECTION_TIMEOUT));
+
+            Assert.assertTrue(
+                    "Failed to invoke zks#createSession() method during client session creation",
+                    simplezks.waitForSessionCreation(5));
+            watcher.waitForConnected(ClientBase.CONNECTION_TIMEOUT);
+            zkClient.close();
+        } finally {
+            // reset cnxn factory
+            if (originalServerCnxnFactory == null) {
+                System.clearProperty(
+                        ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY);
+                return;
+            }
+            System.setProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY,
+                    originalServerCnxnFactory);
+        }
+    }
+
+    /**
+     * Test case for
+     * {@link https://issues.apache.org/jira/browse/ZOOKEEPER-2383}.
+     */
+    @Test(timeout = 30000)
+    public void testFourLetterWords() throws Exception {
+        startSimpleZKServer(startupDelayLatch);
+        verify("conf", ZK_NOT_SERVING);
+        verify("crst", ZK_NOT_SERVING);
+        verify("cons", ZK_NOT_SERVING);
+        verify("dirs", ZK_NOT_SERVING);
+        verify("dump", ZK_NOT_SERVING);
+        verify("mntr", ZK_NOT_SERVING);
+        verify("stat", ZK_NOT_SERVING);
+        verify("srst", ZK_NOT_SERVING);
+        verify("wchp", ZK_NOT_SERVING);
+        verify("wchc", ZK_NOT_SERVING);
+        verify("wchs", ZK_NOT_SERVING);
+        verify("isro", "null");
+    }
+
+    private void verify(String cmd, String expected)
+            throws IOException, SSLContextException {
+        String resp = sendRequest(cmd);
+        LOG.info("cmd " + cmd + " expected " + expected + " got " + resp);
+        Assert.assertTrue("Unexpected response", resp.contains(expected));
+    }
+
+    private String sendRequest(String cmd)
+            throws IOException, SSLContextException {
+        return send4LetterWord(HOST, PORT, cmd);
+    }
+
+    private void startSimpleZKServer(CountDownLatch startupDelayLatch)
+            throws IOException {
+        zks = new SimpleZooKeeperServer(tmpDir, tmpDir, 3000,
+                startupDelayLatch);
+        SyncRequestProcessor.setSnapCount(100);
+        final int PORT = Integer.parseInt(HOSTPORT.split(":")[1]);
+
+        servcnxnf = ServerCnxnFactory.createFactory(PORT, -1);
+        Thread startupThread = new Thread() {
+            public void run() {
+                try {
+                    servcnxnf.startup(zks);
+                } catch (IOException | InterruptedException e) {
```

Comment:
Thank you @fpj . I've separated out the catch block and done another commit in PR.